src/core/room.ts

```diff
@@ -129,53 +138,146 @@ export class DefaultRoom implements Room {
     this._logger = logger;
     this._status = new DefaultStatus(logger);
 
+    // This function gets called if release() is called before initialization
+    // starts. It allows for the room to not be initialised at all since it
+    // won't be needed.
+    let stopInitializingFeatures: (() => void) | undefined;
+
+    // This promise is the same as initAfter but it gets rejected if release()
+    // is called before initialization starts. This make sure that room
+    // features will not permanently hang waiting for this promise to resolve.
+    //
+    // This promise is passed down to all features to wait before starting to
+    // create or use any realtime channels.
+    const initFeaturesAfter = new Promise<void>((resolve, reject) => {
```

Comment:
A note here - any async ops as part of feature initialisation will need to be `await` or become synchronous, we need them all to finish before this promise resolves

Reply:
Good point! All feature init is sync now and runs after this promise resolves. And because the channels are now promises they could be async and those are awaited below (lines 240, 252)